lib/getGitData.js

```diff
@@ -1,12 +1,11 @@
-(function (logger, exec, Q) {
+(function (logger, exec, Promise) {
     'use strict';
     module.exports = {
         getCommitId: function (commitId) {
-            var deferred = Q.defer();
-            process.nextTick(function () {
+            return new Promise(function (resolve, reject) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (9)](https://www.codacy.com/app/codacy/node-codacy-coverage/file/630754750/issues/source?bid=2108217&fileBranchId=2304299#l5)